src/scanner/images/index.ts

```diff
@@ -42,17 +42,33 @@ export async function removePulledImages(images: IPullableImage[]): Promise<void
 
 // Exported for testing
 export function removeTagFromImage(imageWithTag: string): string {
-  return imageWithTag.split('@')[0].split(':')[0];
+  const imageParts = getImageParts(imageWithTag);
+  return imageParts.imageName;
 }
 
 // Exported for testing
 export function getImageTag(imageWithTag: string): string {
-  const imageParts: string[] = imageWithTag.split(':');
-  if (imageParts.length === 2) { // image@sha256:hash or image:tag
-    return imageParts[1];
+  const imageParts = getImageParts(imageWithTag);
+  return imageParts.imageTag;
+}
+
+function getImageParts(imageWithTag: string) : {imageName: string, imageTag: string} {
+  // we're matching pattern: <registry:port_number>(optional)/<image_name>(mandatory)@<tag_identifier>(optional):<image_tag>(optional)
+  const regex = /((?:.*(:\d{4})?\/)?(?:[a-z0-9]+))([@|:].+)?/ig;
+
+  const groups  = regex.exec(imageWithTag);
+  
+  if(!groups){
+    throw Error('Image with tag is malformed, cannot extract valid parts');
   }
 
-  return '';
+  const imageTagParts: string[] | undefined = groups[3] ? groups[3].split(':') : undefined;
+
+  return {
+    imageName: groups[1],
+    //checking if tag malformed, valid formats: image@sha256:hash or image:tag
+    imageTag: imageTagParts && imageTagParts.length === 2 ? imageTagParts[1] : ''
+  };
```

Comment:
I find this flow a bit hard to read and understand, might worth a small refactor

Reply:
fair enough! 👍

Reply:
I have refactored it, so hopefully now it's a lot clearer